scripts/Modules/mkw_core.py

```diff
@@ -88,3 +88,53 @@ def get_speed(playerIdx=0):
     xyz = math.sqrt(x ** 2 + y ** 2 + z ** 2)
 
     return speed(x, y, z, xz, xyz)
+
+# Next 3 functions are used for exact finish display
+def getIGT(lap, player):
+    totalTime = [0, 0]
+    totalTime[0] = chase_pointer(
+        classes.getRaceInfoHolder(), [0xC, (player)*0x4, 0x3C, (lap-1)*0xC + 0x5], 'u8')
+    totalTime[1] = chase_pointer(
+        classes.getRaceInfoHolder(), [0xC, (player)*0x4, 0x3C, (lap-1)*0xC + 0x6], 'u8')
+
+    return totalTime
+
+
+def updateExactFinish(lap, player):
+    finish = [0, 0]
+
+    if player == 0:
+        address = 0x800001B0
+    elif player == 1:
+        address = 0x800001F8
+
+    currentMin = getIGT(lap, player)[0]
+    currentSec = getIGT(lap, player)[1] + \
```

Comment:
You shouldn't be calling this function twice. You should capture the return value once. You can change this to the following:
`currentMin, currentSec = getIGT(lap, player)`